src/PactNet.Native/NativeRequestBuilder.cs

```diff
@@ -78,8 +78,8 @@ IRequestBuilderV2 IRequestBuilderV2.WithQuery(string key, string value)
         /// <param name="key">Header key</param>
         /// <param name="value">Header value</param>
         /// <returns>Fluent builder</returns>
-        IRequestBuilderV2 IRequestBuilderV2.WithHeader(string key, string value)
-            => this.WithHeader(key, value);
+        IRequestBuilderV2 IRequestBuilderV2.WithHeader(string key, dynamic value)
```

Comment:
Ah, we'd specifically moved away from `dynamic` because it's very difficult to get that to play nicely with the Rust code. I'd prefer an overload which took an `IMatcher` perhaps? The API is designed specifically for principle of least surprise, and `dynamic` lets you do some crazy stuff where in reality we'd only support `string` and `IMatcher`.

Reply:
Fair point, I'm definitely not a .Net developer :)